libtenzir/builtins/operators/load_tcp.cpp

```diff
@@ -392,7 +394,7 @@ struct connection_manager_state {
           }
           connection->chunks.push(std::move(chunk));
           TENZIR_ASSERT(connection->chunks.size() <= max_queued_chunks);
-          should_read = connection->chunks.size() < max_queued_chunks;
+          should_read = !stop and connection->chunks.size() < max_queued_chunks;
         }
         if (should_read) {
```

Comment:
```suggestion
        if (should_read and not ec) {
```

And then undo the above changes.